config_samples/examples/client.example.toml

```
@@ -0,0 +1,18 @@
+chain_id = "00000000-0000-0000-0000-000000000000"
+
+[account]
+id = "alice@wonderland"
+public_key = "ed01207233BFC89DCBD68C19FDE6CE6158225298EC1131B6A130D1AEB454C1AB5183C0"
+private_key.digest_function = "ed25519"
+private_key.payload = "9ac47abf59b356e0bd7dcbbbb4dec080e302156a48ca907e47cb6aea1d32719e7233bfc89dcbd68c19fde6ce6158225298ec1131b6a130d1aeb454c1ab5183c0"
+
+[api]
+torii_url = "http://127.0.0.1:8080/"
+basic_auth.login = "mad_hatter"
+basic_auth.password = "ilovetea"
+
+[transaction]
+time_to_live = 100_000
+status_timeout = 100_000
```

```suggestion
time_to_live = 100_000
status_timeout = 100_000
```

without a units suffix we have no idea what we're doing. Is it `_ms`? Please add a suffix

---

I don't like that I need to look through documentation to find which unit this field is set to. It's also error prone, I might forget or misremember. Adding a simple unit suffix just relieves user of all that suffering. [This](https://ruudvanasseldonk.com/2022/03/20/please-put-units-in-names) article sums it up nicely

---

```suggestion
time_to_live = 100_000s
status_timeout = 100_000s
```

I would also accept, although I find it more cumbersome

---

also I don't think `status_timeout` should be only for transactions, could we rename it to something like `request_timeout`?

---

I like the "human-readable string" way the most. Disallowing bare numbers down the line is also good IMO

---

>  # means ms, not clear, disallow?
> duration = 100_000

yes, absolutely should be disallowed

> # human-readable string
> duration = "5m 3s 51ms"

This or units in the type like `duration_ms`. I prefer the 2nd because it's simple and straightforward but I think I would be outvoted on this in the poll. Can we make a poll?

---

I'll make a poll

---

Actually, I raised this topic multiple times already:

- In the RFC itself: [Human-Readable Types](https://github.com/hyperledger/iroha-rfcs/blob/main/text/001-config-overhaul.md#proposal-5---human-readable-types)
- In the Reference tracking issue: https://github.com/hyperledger/iroha-2-docs/issues/392#issuecomment-1708122759

The RFC was reviewed and approved by a wide range of people (I hope so). And in the reference tracking issue, @Mingela and @6r1d had preferred human readable string over a `_ms` format.

Shall we still create a poll?

Anyway, this feature is to be implemented separately, as it is an ideal candidate for it. Upon merging of this PR, I will create all the follow-up tasks to do.